inference-engine/src/mkldnn_plugin/nodes/extract_image_patches.cpp

```diff
@@ -1,248 +1,425 @@
-    std::string _auto_pad;
+        LayerConfig config;
 
-    static const std::set<size_t> _supported_precisions_sizes;
-};
+        DataConfig inConfig;
+        inConfig.desc = inData->getTensorDesc();
+        config.inConfs.push_back(inConfig);
 
-const std::set<size_t> ExtractImagePatchesImpl::_supported_precisions_sizes = {1, 2, 4, 8};
+        DataConfig outConfig;
+        outConfig.desc = layer->outData[0]->getTensorDesc();
+        outConfig.desc.setPrecision(inConfig.desc.getPrecision());
+        outConfig.desc.setLayout(inConfig.desc.getLayout());
+        config.outConfs.push_back(outConfig);
+
+        config.dynBatchSupport = false;
+        confs.push_back(config);
+    } catch (InferenceEngine::Exception &ex) {
+        errorMsg = ex.what();
+    }
+}
 
-REG_FACTORY_FOR(ExtractImagePatchesImpl, ExtractImagePatches);
 
+StatusCode ExtractImagePatchesImpl::execute(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs, ResponseDesc *resp) noexcept {
+    const char *src_data = inputs[0]->cbuffer().as<const char *>() +
+            inputs[0]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+    char *dst_data = outputs[0]->buffer().as<char *>() +
+            outputs[0]->getTensorDesc().getBlockingDesc().getOffsetPadding();
+    const int64_t dtype_size = inputs[0]->getTensorDesc().getPrecision().size();
+
+    const auto& inDims = inputs[0]->getTensorDesc().getDims();
+    const int64_t IC = inDims[1];
+    const int64_t IH = inDims[2];
+    const int64_t IW = inDims[3];
+
+    const auto& outDims = outputs[0]->getTensorDesc().getDims();
+    const int64_t OB = outDims[0];
+    const int64_t OH = outDims[2];
+    const int64_t OW = outDims[3];
+
+    const int64_t KH = _ksizes[0], KW = _ksizes[1];
+    const int64_t SH = _strides[0], SW = _strides[1];
+    const int64_t RH = _rates[0], RW = _rates[1];
+
+    int64_t pad_left = 0, pad_top = 0;
+    if (!CaselessEq<std::string>()(_auto_pad, "valid")) {
+        const int64_t iwStep = KW + (RW - 1) * (KW - 1);
+        const int64_t ihStep = KH + (RH - 1) * (KH - 1);
+        int64_t PW = (std::ceil(1.f * IW/SW) - 1) * SW + iwStep - IW;
+        int64_t PH = (std::ceil(1.f * IH/SH) - 1) * SH + ihStep - IH;
+
+        int64_t increment_sign = 0;
+        if (CaselessEq<std::string>()(_auto_pad, "same_lower")) {
+            increment_sign = 1;
+        } else if (CaselessEq<std::string>()(_auto_pad, "same_upper")) {
+            increment_sign = -1;
+        }
+
+        if ((PW > 0) && (PW < iwStep)) {
+            pad_left = (PW + increment_sign * (PW % 2) ) / 2;
+        }
+        if ((PH > 0) && (PH < ihStep)) {
+            pad_top = (PH + increment_sign * (PH % 2) ) / 2;
+        }
+    }
+    const int64_t PL = pad_left;
+    const int64_t PT = pad_top;
+
+    const std::vector<int64_t> ostrides = {KH * KW * IC * OH * OW, KW * IC * OH * OW, IC * OH * OW, OH * OW};
+    const std::vector<int64_t> istrides = {IC * IH * IW, IH * IW, IW};
```

Comment:
Lines #333-358 can be moved into the constructor.

Reply:
Ok, I'll move the calculation of pad_left and pad_top to the constructor.